### PR TITLE
added the methods and screens for capturing apiKey and spaceId

### PR DESCRIPTION
**Pull request by HussainKhanSherwani**
https://www.loom.com/share/e140ec8b66184efdb62f4cc02e52e893?sid=b9f271a1-5e1c-483b-bd69-2f6f69200779

### CODE REVIEW

File: src/core/bg/index.js [hunk 1]
```diff
@@ -45,15 +45,10 @@ browser.runtime.onMessage.addListener((message, sender) => {
 	if (message.method.startsWith("tabs.")) {
 		return tabs.onMessage(message, sender);
 	}
-	if (message.method.startsWith("updateLocalStorage")) { 
+	if (message.method.startsWith("dodaoBackground.saveSpaceIdAndApiKey")) {
```
Comment (RobinNagpal) [inline src/core/bg/index.js]: We should add if `message.method.startsWith("dodaoBackground.` and then call dodao-upload script and do this logic in that script as you see it done for `tabs.` above

File: src/core/bg/tabs.js [hunk 2]
```diff
@@ -95,7 +95,7 @@ function sendLocalStorageData() {
 async function onTabUpdated(tabId, changeInfo) {
 
 	if (changeInfo.status == "complete") {
-		sendLocalStorageData();
+		// sendLocalStorageData();
```
Comment (RobinNagpal) [inline src/core/bg/tabs.js]: You had added this? Why are we commenting this out?

File: src/core/content/dodao-content.js [hunk 3]
```diff
@@ -1,4 +1,52 @@
 import { v4 as uuidv4 } from "uuid";
+browser.runtime.onMessage.addListener((message) => { 
+  if (message.method === "dodaoContent.captureApiKey") { 
+    showSpaceIdAndApiScreen();
+  }
+});
+function showSpaceIdAndApiScreen() {
+  const inputs = [
+    {
```
Comment (RobinNagpal) [inline src/core/content/dodao-content.js]: We should already have this code right? for creating of this. modal. Why are we adding new code?

File: src/core/bg/index.js (hunk 1 above)
Comment (HussainKhanSherwani) [reply src/core/bg/index.js]: done

File: src/core/bg/tabs.js (hunk 2 above)
Comment (HussainKhanSherwani) [reply src/core/bg/tabs.js]: uncommented

File: src/core/content/dodao-content.js (hunk 3 above)
Comment (HussainKhanSherwani) [reply src/core/content/dodao-content.js]: yeah you are right ,done

File: src/core/content/dodao-content.js [hunk 4]
```diff
@@ -74,7 +79,7 @@ function showLoginScreen(callbackFunction) {
         return;
```
Comment (RobinNagpal) [inline src/core/content/dodao-content.js:79]: Seem like you are not using anything to format your code

File: src/core/bg/dodao-upload.js [hunk 5]
```diff
@@ -1,4 +1,22 @@
 import html2canvas from "html2canvas";
+
+export async function onMessage(message, sender) {
+  if (message.method.endsWith("saveSpaceIdAndApiKey")) {
+    chrome.storage.local.set({ spaceId: message.spaceId, apiKey: message.apiKey })
```
Comment (RobinNagpal) [inline src/core/bg/dodao-upload.js:5]: Extract this logic in a separate function

Comment (RobinNagpal) [reply src/core/bg/dodao-upload.js:5]: We should also do the validation if the spaceId and apiKey are passed. If not then we can call the same method and pass an error message

**Pull request merged**